sktime/datatypes/_check.py

```diff
@@ -302,16 +302,16 @@ def mtype(
     mtypes_negative = dict()
 
     for m_plus_scitype in m_plus_scitypes:
-        valid, msg, _ = check_is_mtype(
+        valid = check_is_mtype(
             obj,
             mtype=m_plus_scitype[0],
             scitype=m_plus_scitype[1],
-            return_metadata=True,
+            return_metadata=False,
         )
         if valid:
             mtypes_positive += [m_plus_scitype[0]]
         else:
-            mtypes_negative[m_plus_scitype[0]] = msg
+            mtypes_negative[m_plus_scitype[0]] = ""
```

Comment:
Copied from the previous discussion in #4140

> @[hoesler](https://github.com/hoesler) hoesler [2 days ago](https://github.com/sktime/sktime/pull/4140/files#r1092252104)
> @fkiraly This is a TODO. The idea was to reduce computation time of checks by setting return_metadata=False, as metadata is not required here. The problem is, however, that error messages are turned off as well. I think the best would be to always return error messages, but that would require an api change of check_is_mtype. What's your opinion?

> @[fkiraly](https://github.com/fkiraly) fkiraly [2 days ago](https://github.com/sktime/sktime/pull/4140/files#r1092276277) • 
> I see what you're doing - very smart observation imo.
> 
> The "clean" way would be a refactor with deprecation.
> One way to do it would be to introduce a return_msg argument everywhere, which can be True, False, or None (default).
> If None, the current behaviour happens and a deprecation warning is raised. If True, always returns the msg return, independently of return_metadata. If False, never returns the msg arg.
> After a deprecation cycle, the default changes to False.

> @[fkiraly](https://github.com/fkiraly) fkiraly [2 days ago](https://github.com/sktime/sktime/pull/4140/files#r1092277077) • 
> it is nightmarish though since the argument would have to be introduced in all the checker functions.
> (the deprecation logic should sit in some public functions rather than in the checkers)
> 
> I would definitely advise a separate PR for this with only that as the purpose, since PRs should be focused in file footprint, or in scope. It is not possible to focus this in terms of file footprint...
> 
> unless you come up with a better idea.